crates/subspace-verification/src/lib.rs

```diff
@@ -231,7 +233,7 @@ where
     let s_bucket_audit_index = sector_slot_challenge.s_bucket_audit_index();
 
     // Check that proof of space is valid
-    if !PosTable::is_proof_valid(
+    if !<PosTable as SolutionPotVerifier>::is_proof_valid(
```

Comment:
@nazar-pc in the subspace benchmarks, I'm getting an `InvalidProofOfSpace` error here:
- https://github.com/autonomys/subspace/blob/545001966d564775274bf1cfa8084fc3ea1b03f9/crates/pallet-subspace/src/extensions/benchmarking.rs#L87
- https://github.com/autonomys/subspace/blob/545001966d564775274bf1cfa8084fc3ea1b03f9/crates/pallet-subspace/src/extensions/benchmarking.rs#L137

I'm not sure if:
- my changes in this PR have broken verification (I've checked and none of my commits should change the result)
- there's an unexpected incompatibility in the abundance source commit
- the benchmark is wrong in some way, and this code is stricter
- the abundance code doesn't handle unlimited solution ranges
- some other issue

Does anything seem obvious to you?

Reply:
Nothing obvious from a quick look, but I have not audited the code. Try to start a local node and farmer, see if it can produce blocks. Even with changed logic for constructing tables, if challenge is derived the same way, verification should succeed. Something is significantly wrong if verification doesn't pass.

You can also add verification right after proof generation and dump the seed and challenge index for which verification fails. You can then update tests to check that particular seed both here and in abundance repo to see where things do not match.

Reply:
Using the hard-coded [vote.data](https://github.com/autonomys/subspace/blob/c3a22e2a2f5f89a66f7e6492b81cfa33ad3a68a6/crates/pallet-subspace/src/extensions/fixtures/vote.data), which is decoded using [VoteData](https://github.com/autonomys/subspace/blob/c3a22e2a2f5f89a66f7e6492b81cfa33ad3a68a6/crates/pallet-subspace/src/extensions/benchmarking.rs#L25-L36), I get a failure in `ShimTable` (only used for tests) at:
https://github.com/autonomys/subspace/blob/fa44a4613b3b102ba0b98384c42ec76d95925efb/crates/subspace-proof-of-space/src/shim.rs#L37

Which means no solution was found.

This happens because `ab-proof-of-space` inverted the `ShimTable` logic, I've reverted to the current Subspace logic, and the benchmarks pass.

Reply:
Why not changing the vote data to match updated shim implementation then? Also if benchmarks are running shim implementation instead of Chia, that doesn't seem right.

Reply:
> Why not changing the vote data to match updated shim implementation then?

I wasn't in the mood for mucking around with binary files 😂 

But seriously, if we are just going to change to the real implementation in the next PR, there's no point in changing the vote data in this one as well.

> Also if benchmarks are running shim implementation instead of Chia, that doesn't seem right.

I had that thought as well. Happy to move to the real implementation in another PR.

If there's any expensive setup we can put it outside the benchmarked code. (Or in a `OnceCell` if it's very expensive.)

@vedhavyas @jfrank-summit what do you think?